sdk/python/kfp/compiler/_default_transformers.py

```diff
@@ -12,30 +12,45 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import warnings
+from kubernetes import client as k8s_client
+
 from ..dsl._container_op import BaseOp, ContainerOp
 
+
 def add_pod_env(op: BaseOp) -> BaseOp:
-    """Adds pod environment info to ContainerOp.
+  """Adds environment info if the Pod has the label `add-pod-env = true`.
+    """
+  if isinstance(
+      op, ContainerOp
+  ) and op.pod_labels and 'add-pod-env' in op.pod_labels and op.pod_labels[
+      'add-pod-env'] == 'true':
+    return add_kfp_pod_env(op)
+
+
+def add_kfp_pod_env(op: BaseOp) -> BaseOp:
+  """Adds KFP pod environment info to the specified ContainerOp.
     """
-    if isinstance(op, ContainerOp) and op.pod_labels and 'add-pod-env' in op.pod_labels and op.pod_labels['add-pod-env'] == 'true':
-        from kubernetes import client as k8s_client
-        op.container.add_env_variable(
-            k8s_client.V1EnvVar(
-                name='KFP_POD_NAME', 
-                value_from=k8s_client.V1EnvVarSource(
-                    field_ref=k8s_client.V1ObjectFieldSelector(
-                        field_path='metadata.name'
-                    )
-                )
-            )
-        ).add_env_variable(
-            k8s_client.V1EnvVar(
-                name='KFP_NAMESPACE', 
-                value_from=k8s_client.V1EnvVarSource(
-                    field_ref=k8s_client.V1ObjectFieldSelector(
-                        field_path='metadata.namespace'
-                    )
-                )
-            )
-        )
-    return op
\ No newline at end of file
+  if not isinstance(op, ContainerOp):
+    warnings.warn(
+        'Trying to add default KFP environment variables to an Op that is '
+        'not a ContainerOp. Ignoring request.')
+    return op
+
+  op.container.add_env_variable(
+      k8s_client.V1EnvVar(name='KFP_POD_NAME',
+                          value_from=k8s_client.V1EnvVarSource(
+                              field_ref=k8s_client.V1ObjectFieldSelector(
+                                  field_path='metadata.name')))
+  ).add_env_variable(
+      k8s_client.V1EnvVar(name='KFP_NAMESPACE',
+                          value_from=k8s_client.V1EnvVarSource(
+                              field_ref=k8s_client.V1ObjectFieldSelector(
+                                  field_path='metadata.namespace')))
+  ).add_env_variable(
+      k8s_client.V1EnvVar(
+          name='WORKFLOW_ID',
```

Comment:
Do we want to expose the Workflow ID, which is an Argo-specific concept?

We have RUN_ID, which is expected to be portable (it does not point to an Argo Workflow though.).